src/ServiceBusExplorer/Forms/MainForm.cs

```diff
@@ -6370,28 +6372,114 @@ private void ReapplyColors(TreeNode parentNode)
+                {
+                    return (true, topicDescription.MessageCountDetails);
+                }
+                // SubscriptionDescription
+                if (entityDescription is SubscriptionDescription subscriptionDescription)
+                {
+                    return (true, subscriptionDescription.MessageCountDetails);
+                }
+
+                return (true, null);
+            }
+            // NotificationHubDescription
+            if (tag is NotificationHubDescription notificationHubDescription)
+            {
+                return (true, null);
+            }
+            // SubscriptionWrapper
+            if (tag is SubscriptionWrapper subscriptionWrapper)
+            {
+                return (true, subscriptionWrapper.SubscriptionDescription?.MessageCountDetails ?? subscriptionWrapper.TopicDescription?.MessageCountDetails);
+            }
+            // RuleWrapper
+            if (tag is RuleWrapper ruleWrapper)
+            {
+                return (true, ruleWrapper.SubscriptionDescription.MessageCountDetails);
+            }
+            // UrlSegmentWrapper
+            if (tag is UrlSegmentWrapper urlSegmentWrapper)
+            {
+                return (false, null);
+            }
+            // EventGridSubscriptionWrapper
+            if (tag is EventGridSubscriptionWrapper eventGridSubscriptionWrapper)
+            {
+                return (true, null);
+            }
+
+            throw new NotSupportedException($"{tag.GetType().FullName} is not support in {nameof(GetColorDataFromNodeTag)}");
+        }
+
+        static bool ColorIsMatch(TreeNode node, NodeColorInfo nodeColorInfo, MessageCountDetails messageCountDetails)
+        {
+            var nameIsMatch = Regex.IsMatch(node.Name, nodeColorInfo.Text ?? string.Empty);
+            if (!nameIsMatch)
+            {
+                return false;
+            }
+
+            if (messageCountDetails == null)
+            {
+                return true;
+            }
+
+            var countChecks = new List<(long Value, long? Threshold)>
+            {
+                (messageCountDetails.ActiveMessageCount, nodeColorInfo.ActiveMessageCountThreshold),
+                (messageCountDetails.DeadLetterMessageCount, nodeColorInfo.DeadLetterCountThreshold),
+                (messageCountDetails.ScheduledMessageCount, nodeColorInfo.ScheduledMessageCount),
+                (messageCountDetails.TransferMessageCount, nodeColorInfo.TransferMessageCountThreshold),
+                (messageCountDetails.TransferDeadLetterMessageCount, nodeColorInfo.TransferDeadLetterMessageCountThreshold),
+            };
+
+            var checkResults = (
+                from c in countChecks
+                where c.Threshold.HasValue
+                let limitExceeded = c.Value > c.Threshold.Value
+                select limitExceeded
+            ).ToList();
+
+            return !checkResults.Any() || checkResults.Any(x => x);
```

Comment:
No need for 2 LINQ calls here.
```suggestion
            return checkResults.Count == 0 || checkResults.Any(x => x);
```